src/main/java/no/nav/fo/veilarbportefolje/consumer/DialogDataFeedHandler.java

```diff
@@ -37,22 +35,10 @@ public DialogDataFeedHandler(BrukerRepository brukerRepository,
     @Override
     @Transactional
     public void call(String lastEntry, List<DialogDataFraFeed> data) {
-
-        try {
-            timed("feed.dialog.objekt",
-                    () -> {
-                        data.forEach(info -> {
-                            dialogFeedRepository.oppdaterDialogInfoForBruker(info);
-                            indekseringService.indekserAsynkront(AktoerId.of(info.getAktorId()));
-                        });
-                    },
-                    (timer, hasFailed) -> timer.addTagToReport("antall", Integer.toString(data.size()))
-            );
-            brukerRepository.updateMetadata(DIALOGAKTOR_SIST_OPPDATERT, Date.from(ZonedDateTime.parse(lastEntry).toInstant()));
-        } catch(Exception e) {
```

Comment:
Når du fjerner catch-blokken her, er det mulig du bør dobbeltsjekke hva det betyr for transaksjonshåndteringen. Så lenge det har vært en catch her, vil transaksjonen _ikke_ bli rullet tilbake igjen. Det vil i praksis si at hvis koden prosesserer en liste på n elementer, og feiler på element nr. m, vil element nr. 1-m være behandlet ok (og ikke bli rullet tilbake igjen). Metadata vil riktignok ikke bli oppdatert, så disse 1-m elementene vil bli lest inn og behandlet på nytt ved neste anledning.
Når vi fjerner catch-blokken, vil transaksjonen feile, og elementene 1-m vil bli rullet tilbake. Det betyr antagelig ikke så mye i praksis, men det kan medføre usynk mellom hva som ligger i databasen og hva som ligger i indeksen, dersom indeksen har blitt oppdatert på bakgrunn av de tilbakerullede dataene fra databasen.
Apropos indeksering: Jeg er usikker på hvordan den asynkrone indekseringen fungerer her, siden den startes i en transaksjonell metode. Det er mulig den asynkrone indekseringen ikke vil få tak i de oppdaterte dataene før transaksjonen er ferdig, og dermed risikerer å indeksere gamle data. Så det er mulig det er en bug her (som i så fall har vært der lenge). Men jeg kan ikke huske å ha observert noen feil tilknytett dette, så det er mulig at indekseringen skjer i samme transaksjon (og får tilgang til riktige data) selv om den kjøres asynkront? Evt. har vi bare hatt flaks, slik at transaksjonen vanligvis fullføres før indekseringen starter. :) 
Jeg ser at OppfolgingFeedHandler ikke er transaksjonell på samme måte som DialogFeedHandler, så det er mulig at disse oppfører seg ulikt.

Reply:
Bra catcha! Skal undersøke det litt videre, er absolutt kjipt med kortslutning her, men syns ikke pokemon-exception-handling er måten å unngå akkurat det på. Er det ikke bedre å bruke et thread-pool vha CompletebaleFuture eller lignende hvor trådene kan bare kaste så mange exceptions de bare vil?

Er usikker på hvorfor den gjør den indekseringen asynkront selv, burde ikke den gjøres på slutten av transaksjonen? Mistenker en del errors i loggene har sammenheng med akkurat dette, bl.a. den hvor elastic prøver å slette samme person flere ganger fra indeksen

Reply:
Det er sikkert jeg som satte opp at indekseringen er asynkron. Tanken der er at det å trigge indeksering bare er "fire and forget", slik at hverken tidsbruk eller eventuelle feil ved indeksering skal påvirke innlesing og lagring av data fra feeden. Men det er jo et poeng at indekseringen får tak i riktige data :) Egentlig er det vel ingen grunn til å ha transactional rundt denne metoden, for vi trenger vel egentlig ikke å rulle tilbake ferdigbehandlede elementer dersom det skjer en feil? Det går også an å trigge indeksering på slutten av transaksjonen, som du nevner. Men siden denne tar inn aktørid, må vi enten kjøre en ny loop for indeksering eller la den ta inn en liste med aktørider. Det går også an å droppe indeksering her i det hele tatt, og heller sørge for at endringene blir plukket opp i deltaindekseringen. Da må deltaindekseringen ta høyde for endringer i dialogdata også, ikke bare Arena-data.